### PR TITLE
Credit card validation rule

### PR DESCRIPTION
**Pull request by camcima**

### CODE REVIEW

**Comment by RickvdStaaij** [comment]
@camcima Thanks a lot for this addition, but I'm afraid we're not going to accept it. Particle is made to be the smallest possible package, but with supporting the most important validations. For this reason we don't want to have any third party dependencies. Everyone is free to extend the validator to be able to add their own custom validations, like this credit card check for example. It's a nice addition but I don't think this should be in particle by default.

**Comment by berry-langerak** [comment]
@RickvdStaaij let's talk about this ;)

**Comment by berry-langerak** [comment]
As mentioned in #121, the reasoning of having as little dependencies as possible is based on not wanting to be tied into a framework eco-system. @RickvdStaaij will dive into the dependency to get a view on its stability.

Again, thanks for your contribution, I really like to see that you're using it a lot :)

File: src/Rule/CreditCard.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    /**
+     * @param $value
+     *
+     * @return bool
+     */
+    private function validateFormat($value)
+    {
+        if (preg_match('/^4[0-9]{12}(?:[0-9]{3})?$/', $value) === 1) {
```
Comment (berry-langerak) [inline src/Rule/CreditCard.php]: Maybe you could move the regexes to a protected property of this class (`$validationRegexes`?), and loop over them like:

```
foreach ($this->validationRegexes as $regex) {
   if (preg_match($regex, $value)) {
      return true;
   }
}
return false;
```

Would probably be easier to add new ones if necessary ;)

**Comment by berry-langerak** [comment]
I really like this PR and want to merge it, but I have one small comment.

**Comment by camcima** [comment]
Done!

**Comment by RickvdStaaij** [comment]
There are no extra dependencies too. I came to an agreement with @berry-langerak that this is nice to include by default in the package. So, nothing is holding this back now, nice!

Sorry for the long wait, I got sick and then it was off my radar, sorry for that.

Thank you for the PR and fixes!

File: src/Rule/CreditCard.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+     *
+     * @var array
+     */
+    protected $validationRegExps = [
+        '/^4[0-9]{12}(?:[0-9]{3})?$/', // Visa
+        '/^5[1-5][0-9]{14}$/', // Mastercard
+        '/^3[47][0-9]{13}$/', // American Express
+        '/^3(?:0[0-5]|[68][0-9])[0-9]{11}$/', // Diners Club
+        '/^6(?:011|5[0-9]{2})[0-9]{12}$/', // Discover
+        '/^(?:2131|1800|35\d{3})\d{11}$/', // JCB
```
Comment (RickvdStaaij) [inline src/Rule/CreditCard.php]: This looks way cleaner by the way, nice one :)

**Pull request merged**